ddl/partition.go

```diff
@@ -318,6 +318,17 @@ func checkDropTablePartition(meta *model.TableInfo, partName string) error {
 	return errors.Trace(ErrDropPartitionNonExistent.GenWithStackByArgs(partName))
 }
 
+func checkPartitionExist(meta *model.TableInfo, parName string) error {
+	// TODO: MySQL behavior for hash partition is weird, "create table .. partition by hash partition 4",
+	// it use p0, p1, p2, p3 as partition names automatically.
+	for _, def := range meta.Partition.Definitions {
+		if strings.EqualFold(def.Name.L, strings.ToLower(parName)) {
```

Comment:
save `strings.ToLower(partName)` to local variable? ditto in line#399

Reply:
done